src/tpm2_param_enc.c

```diff
@@ -44,10 +45,170 @@
  *
  */
 
+/******************************************************************************/
+/* --- Local Variables -- */
+/******************************************************************************/
+wc_Sha sha1digest;
```

Comment:
Let's not use static global memory. 
Use stack and pass pointer. 

We have wrappers for all this in `wolfssl/wolfcrypt/hash.h`.

```
wc_HashAlg
WOLFSSL_API int wc_HashInit_ex(wc_HashAlg* hash, enum wc_HashType type,
    void* heap, int devId);
WOLFSSL_API int wc_HashInit(wc_HashAlg* hash, enum wc_HashType type);
WOLFSSL_API int wc_HashUpdate(wc_HashAlg* hash, enum wc_HashType type,
    const byte* data, word32 dataSz);
WOLFSSL_API int wc_HashFinal(wc_HashAlg* hash, enum wc_HashType type,
    byte* out);
WOLFSSL_API int wc_HashFree(wc_HashAlg* hash, enum wc_HashType type);
```

Reply:
Most definitely will use the wolfCrypt wrappers. 
The global memory was to make it easier to focus on the KDFa issue.
Will fix this. Thank you for the feedback!

Reply:
Reverted back to the prior implementation using wolfcrypt HMAC